### DISCUSSION TITLE
How I Can Edit Multiple Cells with single input?

### DISCUSSION THREAD

**Discussion by sravaniuppara** [glideapps/glide-data-grid#536] [Q&A] [answered]
I Wanted to select the multiple cells and need to edit selected cells at once.
I am trying using onCellsEdited but it is giving only one cell co-ordinate.
which function is best to use?
![image](https://user-images.githubusercontent.com/116088886/196457061-fb7071be-6032-4ee6-8ae4-8f3071fee6db.png)

**Comment by jassmith** [comment, marked as answer]
What do you expect to have happen? All cells get the new value? What happens if they are different types?

**Reply by sravaniuppara** [reply]
All the Cells need to get the new value.

**Reply by sravaniuppara** [reply]
All the Selected cells need to be updated with the new Value(multiple characters).

**Reply by sravaniuppara** [reply]
which inBuild function i need to use for mutliCellEdit.

**Reply by jassmith** [reply]
This is not something GDG currently does by default. In most applications this is not the desired behavior. What you want to do is the following.

1) Make sure you are passing the `gridSelection` and `onGridSelectionChange` props.
2) When you receive an onCellEdited event check your `gridSelection` and update not just `gridSelection.current.cell` but all cells in `gridSelection.current.range` with the new value.
3) Make sure you pass every cell in the `range` to `DataGridRef.updateCells`
